src/compiler/printer/unique.ts

```diff
@@ -0,0 +1,65 @@
+import { UniqueIdentifierFlags } from '../ast/node';
+import { UniqueIdentifier } from '../ast/internal/unique-identifier';
+import { Char } from '../parser/scanner/char';
+import { lastOrUndefined } from '../parser/common';
+import { Printer } from './formatter';
+
+// Flags enum to track count of temp variables and a few dedicated names
+export const enum TempFlags {
+  Auto = 0x00000000, // No preferred name
+  CountMask = 0x0fffffff, // Temp variable counter
+  _i = 0x10000000 // Use/preference flag for '_i'
+}
+
+export function generateUniqueName(node: UniqueIdentifier, printer: Printer): string {
+  return (
+    printer.autoGeneratedIdToGeneratedName[node.uniqueId!] ||
```

Comment:
@aladdin-add Same here. **delete this file**

Reply:
done